corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -343,20 +343,35 @@ hqDefine("cloudcare/js/formplayer/app", function () {
                 false
             );
         }
+
+        const pendingErrors = [];
```

Comment:
Do we need an array here? Because it is listed as an array, I'm trying to work through the logic to see if it's possible to have multiple offline notifications get scheduled, at which point coming online would not un-schedule all those notifications. If it's an array to support future notifications, then it opens the door to popping off the wrong scheduled notification.  However, if we do stick with the array, I think it could use a name change -- these are not errors, they are notification timer IDs.

Reply:
I'm also a bit confused about the array. Could this just be a boolean? Turn it on when the user goes offline, then two seconds later check to see if the user is still offline: if they are, show the message, if they're back online, just turn the flag off.